src/plugins/explore/public/components/results_summary/results_summary_panel.tsx

```diff
@@ -138,97 +91,73 @@ export const ResultsSummaryPanel: React.FC<ResultsSummaryPanelProps> = (props) =
-      updateQueryState({ question: '', query: '' });
-    };
-  }, [updateQueryState]);
-
-  useEffect(() => {
-    const subscription = search.df.df$
-      .pipe(
-        distinctUntilChanged(),
-        filter((value) => !isEmpty(value) && !isEmpty(value?.fields))
-      )
-      .subscribe((df) => {
-        if (df) {
-          setResults(convertResult(df));
-        }
-      });
-    return () => {
-      subscription.unsubscribe();
-    };
-  }, [search.df.df$]);
-
   const fetchSummary = useCallback(
-    async (queryContext: QueryContext) => {
+    async ({ question, query, queryRes }: { question: string; query: string; queryRes: any }) => {
       setSummary('');
-      if (isEmpty(queryContext?.queryResults)) {
+      if (isEmpty(queryRes)) {
         return;
       }
 
       setLoading(true);
       setFeedback(FeedbackStatus.NONE);
 
-      const SUCCESS_METRIC = 'generated';
+      const successMetric = 'generated';
       try {
-        const actualSampleSize = Math.min(sampleSize, queryContext?.queryResults?.length);
-        const dataString = JSON.stringify(queryContext?.queryResults?.slice(0, actualSampleSize));
+        const actualSampleSize = Math.min(SUMMARY_REQUEST_SAMPLE_SIZE, queryRes?.length);
+        const dataString = JSON.stringify(queryRes?.slice(0, actualSampleSize));
         const payload = `'${dataString}'`;
+        // TODO: OSD core should not rely on plugin APIs, refactor this once this RFC is
+        // implemented #9859
         const response = await http.post('/api/assistant/data2summary', {
           body: JSON.stringify({
             sample_data: payload,
             sample_count: actualSampleSize,
-            total_count: queryContext?.queryResults?.length,
-            question: 'Show all the logs', // TODO
-            ppl: queryContext?.query,
+            total_count: queryRes?.length,
+            question,
+            ppl: query,
           }),
           query: {
-            dataSourceId: selectedDataset.current?.dataSource?.id,
+            dataSourceId: dataSetState?.dataSource?.id,
           },
         });
         setSummary(response);
-        reportCountMetric(SUCCESS_METRIC, 1);
+        reportCountMetric(successMetric, 1);
       } catch (error) {
-        reportCountMetric(SUCCESS_METRIC, 0);
-        setSummary(errorPrompt);
+        reportCountMetric(successMetric, 0);
+        setSummary(
+          i18n.translate('explore.resultsSummary.summary.errorPrompt', {
+            defaultMessage: 'I am unable to respond to this query. Try another question.',
+          })
+        );
       } finally {
         setLoading(false);
       }
     },
-    [http, reportCountMetric, errorPrompt]
+    [http, reportCountMetric, dataSetState?.dataSource?.id]
   );
 
   useEffect(() => {
-    if (accordionState === 'open' && canGenerateSummary && !summary && !loading) {
-      // TODO properly integrate with t2ppl
+    if (
+      accordionState === 'open' &&
```

Comment:
The `summary` here is the last result summary, does it mean the result summary will only be triggered when last result `summary` is empty? For example:
1. I run a natural language query
2. result summary is generated
3. Now I run another natural language query
4. Will it trigger new result summary for the new query?

Reply:
The component actually re-renders if the query is executed so the summary will not be present. The current implementation does trigger when we re-run the query: 

https://github.com/user-attachments/assets/42518cc6-74cc-4d7e-9eb2-2ff916be13ae